core/domain/prod_validation_jobs_one_off.py

```diff
@@ -648,13 +747,38 @@ def _validate_explorations_are_public(cls, item):
                     item.id, private_exp_ids))
 
     @classmethod
-    def _validate_collections_are_public(cls, item):
+    def _validate_collections_are_public(
+            cls, item, field_name_to_external_model_references):
         """Validates that collections for model are public.
 
         Args:
             item: ndb.Model. BaseUserModel to validate.
+            field_name_to_external_model_references:
+                dict(str, (list(ExternalModelReference))).
+                A dict is keyed by field name. Each value consists of a list
```

Comment:
Ditto, again.

I'll stop commenting on these -- please fix throughout.

I am also wondering -- for this one at least -- why not just pass field_name_to_external_model_references['collection_ids']? It would make it easier to understand this method and the other fields are superfluous anyway. Similar comments apply to other methods.

Reply:
Updated. 

This function is called for different validators through the base validator. For different validators we need to validate one or more external models and so we cannot pass only a specific id since the call is made [here](https://github.com/oppia/oppia/pull/9495/files#diff-ecafbbaeb117bcae187e78e5b33e120fR369) for all such functions (which use the field_name_to_external_model_references dict) and we cannot use a single call if we make this dependent on ids.